ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -194,23 +150,15 @@ internal constructor(
     // TODO: Since we depend on survey stream from repo anyway, this transformation can be moved
     // into the repo
     // LOIs that are persisted to the local and remote dbs.
-    val loiStream =
-      surveyRepository.activeSurvey.switchMap { survey ->
-        activeSurveyId = survey.map { it.id }.orElse("")
-        getLocationsOfInterestStream(survey)
-      }
-
-    val savedMapLocationsOfInterest =
-      Flowable.combineLatest(
-        loiStream.map { locationsOfInterest -> toLocationOfInterestFeatures(locationsOfInterest) },
-        selectedLocationOfInterest
-      ) { locationsOfInterest, selectedLocationOfInterest ->
-        updateSelectedLocationOfInterest(locationsOfInterest, selectedLocationOfInterest)
-      }
 
+    // TODO: Should we only render the LOIs that are present within the map bounds?
```

Comment:
This is a bigger project imo and not actionable here. Can you move this to an issue instead of a TODO?